app/models/purchase/line_item.rb

```diff
@@ -2,33 +2,15 @@ class Purchase::LineItem < ActiveRecord::Base
   belongs_to :purchase_goods, :class_name => 'Purchase::Good', foreign_key: :good_id
   belongs_to :purchase_suppliers, :class_name => 'Purchase::Supplier', foreign_key: :supplier_id
   belongs_to :purchase_purchases, :class_name => 'Purchase::Purchase', foreign_key: :purchase_id
+  has_many :purchase_contract_items, :class_name => 'Purchase::ContractItem', foreign_key: :line_item_id, dependent: :destroy
 
-  validates :good_id, uniqueness: { scope: :purchase_id,
-                                   message: 'Товар не должен повторяться' }
+  accepts_nested_attributes_for :purchase_contract_items
+
+  validates_uniqueness_of :good_id, message: 'Товар не должен повторяться'
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.